safe_transaction_service/events/services/queue_service.py

```diff
@@ -0,0 +1,272 @@
+        self._channel = None
+        if self._connection and self._connection.is_open:
+            # If channel was closed and connection is still active we try to reopen the channel
+            logger.error(
+                "Connection is opened retry to open channel in 5 seconds: %s",
+                self._connection_parameters.host,
+                reason,
+            )
+            self._connection.ioloop.call_later(5, self.open_channel())
+
+    def setup_exchange(self):
+        """Setup the exchange on RabbitMQ by invoking the Exchange.Declare RPC
+        command. When it is complete, the on_exchange_declareok method will
+        be invoked by pika.
+
+        """
+        logger.info("Declaring exchange %s", self.EXCHANGE_NAME)
+
+        self._channel.exchange_declare(
+            exchange=self.EXCHANGE_NAME,
+            exchange_type=ExchangeType.fanout,
+            durable=True,
+            callback=self.on_exchange_declareok,
+        )
+
+    def on_exchange_declareok(self, _unused_frame):
+        """Invoked by pika when RabbitMQ has finished the Exchange.Declare RPC
+        command.
+        Send unsent messages that cannot be sent as due connection errors.
+
+        :param pika.Frame.Method unused_frame: Exchange.DeclareOk response frame
+        """
+
+        logger.info("Exchange declared: %s", self.EXCHANGE_NAME)
+        self.send_unsent_events()
+
+
+class SyncQueueService(QueueService):
+    """
+    Synchronous connection with test purpose as we cannot test using gevent connection
+    """
+
+    def __init__(self):
+        super().__init__()
+        self.connect()
+
+    def connect(self):
+        """
+        This method connects to RabbitMq using Blockingconnection.
+        Store in _connection the BlocingConnection object and creates a new channel
+
+        """
+        try:
+            self._connection = BlockingConnection(self._connection_parameters)
+            self._channel = self.open_channel()
+            self.setup_exchange()
+        except pika.exceptions.AMQPConnectionError:
+            logger.error("Cannot open connection, retrying")
+
+    def open_channel(self) -> Channel:
+        """
+        Open a new channel
+
+        :return: channel opened
+        """
+        return self._connection.channel()
+
+    def setup_exchange(self):
+        """Setup the exchange on RabbitMQ by invoking the Exchange.Declare RPC
+        command.
+
+        """
+        logger.info("Declaring exchange %s", self.EXCHANGE_NAME)
+
+        self._channel.exchange_declare(
+            exchange=self.EXCHANGE_NAME, exchange_type=ExchangeType.fanout, durable=True
+        )
+
+
+class MockedQueueService:
+    """
+    Mocked class to use in case that there is not rabbitMq queue to send events
+    """
+
+    def send_event(self, event: Dict[str, Any]):
+        logger.warning("MockedQueueService: Not sending event with payload %s", event)
```

Comment:
I would make this debug, as people with this not configured should not get a warning